python/cudf/cudf/core/index.py

```diff
@@ -389,13 +567,38 @@ def to_array(self, fillna=None):
         """
         return self._values.to_array(fillna=fillna)
 
-    def to_series(self):
+    def to_series(self, index=None, name=None):
+        """
+        Create a Series with both index and values equal to the index keys.
+        Useful with map for returning an indexer based on an index.
+
+        Parameters
+        ----------
+        index : Index, optional
+            Index of resulting Series. If None, defaults to original index.
+        name : str, optional
+            Dame of resulting Series. If None, defaults to name of original
+            index.
+
+        Returns
+        -------
+        Series
+            The dtype will be based on the type of the Index values.
+        """
+
         from cudf.core.series import Series
 
-        return Series(self._values)
+        return Series(
+            self._values,
+            index=self.copy() if index is None else index,
```

Comment:
Did you confirm that this is actually a deep copy of the input index in Pandas?

Reply:
Yikes, this is actually creating a shallow copy: https://github.com/pandas-dev/pandas/blob/v1.0.3/pandas/core/indexes/base.py#L1111